### PR TITLE
Fix wrong tasks for tau-retail

### PR DESCRIPTION
**Pull request by SeungyounShin**
First and foremost, we want to express our sincere gratitude for creating such an excellent benchmark. The `tau-retail` test dataset is a valuable resource for validating methodologies in the field.

During our evaluation process, we identified some issues that we believe need addressing to enhance the benchmark's reliability and robustness. Specifically:

1. Some examples contain incorrect r_actions (recommended actions).
2. Certain user instructions are ambiguous or miswritten and need clarification.

This pull request aims to rectify these issues, improving the quality and accuracy of the test dataset. These enhancements will lead to more reliable model evaluations, ultimately benefiting the entire research community.

Changes made:
- Corrected r_actions in examples where they were incorrect
- Clarified ambiguous user instructions to make them more specific and clear

We believe these issues, if left unaddressed, could affect the robustness and reliability of the benchmark. 

Related issue: #3

We kindly request a review of these changes. If you notice any other examples that need correction or clarification, please let us know, and we'll address them in this PR or create a follow-up task.

Thank you for considering our contribution to this important benchmark.

@SamWarmuth @karthikncode @pedramr @ysymyth

### CODE REVIEW

File: tau_bench/envs/retail/tasks.py [hunk 1]
```diff
@@ -167,7 +167,7 @@
                 "arguments": {
                     "order_id": "#W6390527",
                     "item_ids": ["8538875209"],
-                    "payment_method_id": "paypal_9497703",
+                    "payment_method_id": "paypal_7644869",
```
Comment (SeungyounShin) [inline tau_bench/envs/retail/tasks.py:170]: For order W6390527, the payment_id in r_action is set to paypal_9497703, but this belongs to a different user's card. The original payment card for order W6390527 is paypal_7644869, which is also the user's default payment_id.

File: tau_bench/envs/retail/tasks.py [hunk 2]
```diff
@@ -2269,7 +2269,7 @@
                 },
             }
         ],
-        "instruction": "You name is Yusuf Hernandez and your email is yusuf.hernandez8836@example.com. You are shy, rigid. You want to exchange your Fleece Jacket red color and half zipper.",
+        "instruction": "You name is Yusuf Hernandez and your email is yusuf.hernandez8836@example.com. You are shy, rigid. You want to exchange your Fleece Jacket for a large red Fleece Jacket with a half zipper",
```
Comment (SeungyounShin) [inline tau_bench/envs/retail/tasks.py:2272]: The statement of facts is incorrect. "Fleece Jacket red color and half zipper" was meant to describe what the user already owns - a red jacket with a half zipper. As we can see in the image, the user simulator also mistakenly talks about this as if it's what they already have. If we look at the r_action and synthetic instruction, the original correct answer is that the user wants to exchange their Fleece Jacket for one with a red color and half zipper.

스크린샷 2024-07-10 오후 3 45 24

**Review by noahshinn — APPROVED** [review]
Thank you for catching these mistakes!

**Comment by ysymyth** [comment]
Thanks a lot for trying our benchmark and catching mistakes! 

Given we'll continuously fix issues, maybe we should start to version the benchmark? So that numbers are only comparable if they are versioned? What do you guys think

**Pull request merged**